pac4j-saml/src/main/java/org/pac4j/saml/metadata/MetadataResolverFactory.java

```diff
@@ -0,0 +1,10 @@
+package org.pac4j.saml.metadata;
+
+import org.opensaml.saml.metadata.resolver.impl.AbstractBatchMetadataResolver;
+import org.springframework.core.io.Resource;
+
+public interface MetadataResolverFactory {
+
+	AbstractBatchMetadataResolver getInstance(Resource metadataResource) throws Exception;

```

Comment:
Please use the parent interface, `MetadataResolver`, if possible as the return type.